### PR TITLE
Add type annotations for files in /utils

### PR DESCRIPTION
**Pull request by nicarl**
### What does this changes

I added type annotations for the functions in utils.

### What was wrong

In order to introduce type checking I suggested running mypy on the codebase (#1334 ). This is a first attempt to cover a small part of the codebase. 
Benefit for developers should be instantaneous as most IDEs already give some hints.

### How this fixes it

Adding type annotations for function arguments and return values of functions.

### CODE REVIEW

**Comment by nicarl** [comment]
It seems that variable annotation (in contrast to annotations of function arguments and return values) were introduced in python 3.6.
I guess you want to keep support for python 3.5? One approach could be to just annotate functions for now.

File: faker/utils/loading.py [hunk 1]
```diff
@@ -37,25 +39,26 @@ def list_module(module):
         return [name for _, name, is_pkg in pkgutil.iter_modules([path]) if is_pkg]
 
 
-def find_available_locales(providers):
-    available_locales = set()
+def find_available_locales(providers: List[str]) -> List[str]:
+    available_locales: Set[str] = set()
```
Comment (nicarl) [inline faker/utils/loading.py:43]: >= python 3.6 required for variable annotations :-(

**Comment by fcurella** [comment]
We've just deprecated 3.5 with #1332

**Comment by nicarl** [comment]
I took the liberty to also update the config file for appveyor.

**Comment by fcurella** [comment]
Thank you! ✨

**Pull request merged**